### PR TITLE
[processor/tailsamplingprocessor] Add a max_spans configuration for the span_count policy of tailsamplingprocessor

### PR DESCRIPTION
**Pull request by mizhexiaoxiao**
**Description:** <Describe what has changed.>
Add a max_spans configuration for the span_count policy of tailsamplingprocessor

**Link to tracking Issue:** <https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18215>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: processor/tailsamplingprocessor/internal/sampling/span_count_sampler.go
Comment (jpkrohling) [inline processor/tailsamplingprocessor/internal/sampling/span_count_sampler.go:45]: You are potentially casting the spanCount to int multiple times, and there's never a situation you are not casting it. Why not just cast it on line 43 directly?

File: processor/tailsamplingprocessor/README.md
Comment (jpkrohling) [inline processor/tailsamplingprocessor/README.md:25]: I think it should mention that the boundaries are inclusive. Like, max_spans=20 will return spans with 20, but not 21.

File: processor/tailsamplingprocessor/internal/sampling/span_count_sampler.go
Comment (TylerHelmuth) [inline processor/tailsamplingprocessor/internal/sampling/span_count_sampler.go:49]: Do we need the other cases?  Does this case catch everything within the range if we use `> min` and `<= max`?

File: processor/tailsamplingprocessor/internal/sampling/span_count_sampler.go
Comment (mizhexiaoxiao) [reply processor/tailsamplingprocessor/internal/sampling/span_count_sampler.go:45]: Yes, it would be better

File: processor/tailsamplingprocessor/internal/sampling/span_count_sampler.go
Comment (mizhexiaoxiao) [reply processor/tailsamplingprocessor/internal/sampling/span_count_sampler.go:49]: Other cases are for the user to only set one of `min_spans` and `max_spans` config options. Assuming that only `min_spans` is set, `max_spans` will be empty value 0. In this case, it is wrong to use `>=min` and `<=max`

File: processor/tailsamplingprocessor/README.md
Comment (mizhexiaoxiao) [reply processor/tailsamplingprocessor/README.md]: UPDATE mentions inclusion of minimum and maximum values

**Review by TylerHelmuth — APPROVED** [review]
LGTM

**Comment by yehaotian** [comment]
Thanks for the work, any timeline this can be released?

**Comment by jiaojialin** [comment]
Hello , @jpkrohling could you please take another look? this is a very useful feature that could make a difference immediately. Would love to have it ASAP. thanks

**Comment by jiaojialin** [comment]
Hello, this PR is pretty simple and straightforward,  could someone please take another look and approve it? Thanks cc @jpkrohling

**Review by TylerHelmuth — COMMENTED** [review]
@mizhexiaoxiao please take a look at the merge conflicts

**Review by jpkrohling — APPROVED** [review]

**Comment by jpkrohling** [comment]
Sorry for the delay in reviewing this. LGTM, this will be merged once the merge conflicts are solved and CI is green.

**Comment by jiaojialin** [comment]
Thanks so much @mizhexiaoxiao @TylerHelmuth @jpkrohling !   Looking forward to this feature the be released!   @mizhexiaoxiao could you please resolve the conflicts at your earliest convenience and merge it? thanks so much!!

**Comment by mizhexiaoxiao** [comment]
Ready to merge!

**Pull request merged**
